### PR TITLE
Refactor BSplineSurface3D periodicity to a more efficient and precise method

### PR DESCRIPTION
**Pull request by GabrielJMS**
* **Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines (explicit, in English)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bug fix: Refactoring BSplineSurface periodicity
- [ ] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:

### CODE REVIEW

File: volmdlr/faces.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 from geomdl import utilities
 from geomdl.fitting import interpolate_surface, approximate_surface
 from geomdl.operations import split_surface_u, split_surface_v
+from geomdl import construct
```
Comment (AxelRinghausen) [inline volmdlr/faces.py:25]:
```suggestion
from geomdl.construct import extract_curves
```

File: volmdlr/faces.py [hunk 2]
```diff
@@ -2230,6 +2231,7 @@ def __init__(self, degree_u, degree_v, control_points, nb_u, nb_v,
         # surface_points = surface.evalpts
 
         self.surface = surface
+        self.curves = construct.extract_curves(surface, extract_u=True, extract_v=True)
```
Comment (AxelRinghausen) [inline volmdlr/faces.py:2234]:
```suggestion
        self.curves = extract_curves(surface, extract_u=True, extract_v=True)
```

**Review by AxelRinghausen — CHANGES_REQUESTED** [review]

File: volmdlr/faces.py [hunk 3]
```diff
@@ -2309,25 +2329,45 @@ def __init__(self, degree_u, degree_v, control_points, nb_u, nb_v,
...
+            self._bbox = self._bounding_box()
+        return self._bbox
+
+    def _bounding_box(self):
+        """
+        Computes the bounding box ot the surface.
+
+        This method is not exact!
+        """
+        return volmdlr.core.BoundingBox.from_points(self.control_points)
```
Comment (AxelRinghausen) [inline volmdlr/faces.py:2370]: Consider using geomdl : https://nurbs-python.readthedocs.io/en/5.x/module_abstract.html#geomdl.abstract.Surface.bbox

**Review by AxelRinghausen — CHANGES_REQUESTED** [review]

File: volmdlr/faces.py (hunk 3 above)
Comment (GabrielJMS) [reply volmdlr/faces.py]: Done. Thank you!

**Review by AxelRinghausen — APPROVED** [review]

**Pull request merged**